lib/contractor_list/bloc/search_state.dart

```diff
@@ -0,0 +1,51 @@
+part of 'search_bloc.dart';
+
+enum SearchStateStatus { initial, loading, failure, success, invalid, valid }
+
+class SearchState extends Equatable {
+  final SearchStateStatus status;
+  final bool isButtonOn;
+  final String query;
+  final String errormsg;
+  final List<Contractor> contractors;
+
+  const SearchState(
+      {required this.isButtonOn,
```

Comment:
isButtonOn is redundant. You can determine this by just checking if the query is empty in the state